packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessage.ts

```diff
@@ -1392,24 +1394,20 @@ export class PaginatedMessage {
 		const embedsWithFooterApplied = deepClone(message.embeds);
 
 		const idx = embedsWithFooterApplied.length - 1;
-		const lastEmbed = embedsWithFooterApplied[idx];
-		if (lastEmbed) {
-			const jsonTemplateEmbed = isJSONEncodable(this.template.embeds?.[idx] ?? this.template.embeds?.[0])
-				? ((this.template.embeds?.[idx] ?? this.template.embeds?.[0]) as JSONEncodable<APIEmbed> | undefined)?.toJSON()
-				: ((this.template.embeds?.[idx] ?? this.template.embeds?.[0]) as APIEmbed | undefined);
+		if (embedsWithFooterApplied.length > 0) {
+			let lastEmbed = embedsWithFooterApplied[idx];
+			const templateEmbed = this.template.embeds?.[idx] ?? this.template.embeds?.[0];
+			const jsonTemplateEmbed = isJSONEncodable(templateEmbed) ? templateEmbed.toJSON() : templateEmbed;
 
 			if (isJSONEncodable(lastEmbed)) {
-				const casted = lastEmbed as EmbedBuilder;
-				casted.data.footer ??= { text: jsonTemplateEmbed?.footer?.text ?? '' };
-				casted.data.footer.text = `${this.pageIndexPrefix ? `${this.pageIndexPrefix} ` : ''}${index + 1} / ${this.pages.length}${
-					casted.data.footer.text ? ` ${this.embedFooterSeparator} ${casted.data.footer.text}` : ''
-				}`;
-			} else {
-				lastEmbed.footer ??= { text: jsonTemplateEmbed?.footer?.text ?? '' };
-				lastEmbed.footer.text = `${this.pageIndexPrefix ? `${this.pageIndexPrefix} ` : ''}${index + 1} / ${this.pages.length}${
-					lastEmbed.footer.text ? ` ${this.embedFooterSeparator} ${lastEmbed.footer.text}` : ''
-				}`;
+				lastEmbed = lastEmbed.toJSON();
+				embedsWithFooterApplied[idx] = lastEmbed;
 			}
+
+			lastEmbed.footer ??= { text: jsonTemplateEmbed?.footer?.text ?? '' };
+			lastEmbed.footer.text = `${this.pageIndexPrefix ? `${this.pageIndexPrefix} ` : ''}${index + 1} / ${this.pages.length}${
+				lastEmbed.footer.text ? ` ${this.embedFooterSeparator} ${lastEmbed.footer.text}` : ''
+			}`;
 		}
```

Comment:
Why is this impl completely changed. Footer data is being set completely differently now. Probably incorrectly. Either revert or provide evidence of testing every single edge case / scenario / code branch.

Reply:
It's changed because you shouldn't rely on `.data` (it's in fact semi-private). Instead, it's converted to the raw type (and obviously replaced in the `embedsWithFooterApplied` array since they're different object references).

Once the data is normalised into what Discord expects, we can do the rest.

The if/else duplicated every line of code, with the difference being `lastEmbed.data.*` vs `lastEmbed.*`, because we normalise the embed now, that's not longer an issue.

I can test this later, of course.

Reply:
What needs testing
1. A page with 1 embed
2. A page with multiple embeds (page index only on the last one)
3. 1 and 2 but with a template without footer
4. 1 and 2 but with a template with a footer, but only an icon
5. 1 and 2 but with template with a footer with both icon and text